workers/unity/Packages/com.improbable.gdk.core/Config/LocatorConfig.cs

```diff
@@ -33,16 +34,65 @@ public LocatorConfig()
         }
 
         /// <summary>
-        ///     Checks that the LocatorConfig instance is valid. This does not guarantee a successful connection.
+        ///    Checks that the connection configuration is valid. This does not guarantee a successful connection.
         /// </summary>
-        public override void Validate()
+        /// <param name="errorMessage">Reason for failing the validation in case false is returned.</param>
+        /// <returns>True fo the connection configuration is valid.</returns>
+        public override bool Validate(out string errorMessage)
         {
-            ValidateConfig(LocatorHost, RuntimeConfigNames.LocatorHost);
+            if (string.IsNullOrEmpty(LocatorHost))
+            {
+                errorMessage = string.Format(MissingConfigError, RuntimeConfigNames.LocatorHost);
+                return false;
```

Comment:
Optional improvement (this'd diverge from previous behaviour though): If you have multiple errors it will only return the first failure... not too helpful in that case... but it may need to output a list (or concat into a string)?

Reply:
Yeah, this is a good point. The thing though is that in areas with branching logic (see the `Validate()` method for `LocatorConfig`) it can be hard to assemble a full set of errors, i.e. if the credentialstype is invalid then it's undefined whether the subsequent checks should pass or fail.

Reply:
I see, np